boards/arduino-leonardo/leonardo-runner.sh

```diff
@@ -1,6 +1,39 @@
 #!/usr/bin/env sh
 set -e
 
+LINUX_OS="LINUX"
+MAC_OS="MAC"
+UNKNOWN_OS="UNKNOWN"
+
+case "$(uname -s)" in
+    Linux*)     OS=$LINUX_OS;;
+    Darwin*)    OS=$MAC_OS;;
+    *)          OS=$UNKNOWN_OS
+esac
+
+if ! command -v numfmt &> /dev/null
+then
+    echo "required numfmt could not be found!"
+    echo "please install https://command-not-found.com/numfmt"
+    exit 1
+fi
+
+if ! command -v avrdude &> /dev/null
+then
+    echo "required avrdude could not be found!"
+    echo "please install https://command-not-found.com/avrdude"
```

Comment:
Error messages should go to `stderr`:

```suggestion
    echo "required avrdude could not be found!" >&2
    echo "please install https://command-not-found.com/avrdude" >&2
```